### PR TITLE
Fix #622 Redesign of favourite menu

### LINKED ISSUE #622: Redesign of the Favourite Menu
## 1.Onclick folder issue:- 
Favourite folders are not clickable
 ## 2. Action bar issue:-
If we add favourite folders then again we get favourite button on the actionbar (although we are still in actionbar)
## 3. Shared Preference issue:-
Shared peferences is not used properly in showing the favourite list(once we close the fragment all the favourite list are disappeared)
## 4. UI of selecting favourite folders:-
View of selecting the favourite folders from the recycler view is not good(although shared preference is used. But there should be a done button after selecting the checkbox)

### Would you like to work on the issue?
Yes

### PR DESCRIPTION
**Pull request by theabhishekavi**
# Description

1. Onclick issue is solved
2. SharedPreferences issue is solved
3. UI of favourite folders is improved.

UI after the change:-

image

Fixes #622 

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by vyankatesh24** [comment]
@theabhishekavi 
* The items getting added to favourite without clicking on `Done`. (when clicked back also they are getting added).

* TextToPDF is not getting added to `Favourites`

**Comment by vyankatesh24** [comment]
* Add Images Icon is White make it black

File: app/src/main/res/xml/favourites_preferences.xml [hunk 1]
```diff
@@ -3,73 +3,96 @@
 
     <CheckBoxPreference
         android:title="Images to PDF"
-        android:key="Images to PDF"/>
+        android:key="Images to PDF"
```
Comment (vyankatesh24) [inline app/src/main/res/xml/favourites_preferences.xml]: Use String from strings.xml

Comment (vyankatesh24) [inline app/src/main/res/xml/favourites_preferences.xml]: Use String from strings.xml

Comment (theabhishekavi) [reply app/src/main/res/xml/favourites_preferences.xml]: Okay!

**Comment by theabhishekavi** [comment]
@vyankatesh24 i will make the add images icon to be black and add TextToPDF to favourites.  But what should be done when user click backbutton instead of done button? Just uncheck all the checkbox?

**Comment by theabhishekavi** [comment]
@vyankatesh24 with due respect, according to me either user click the done button or back button whatever is the condition of checkbox that should remain the same. If we uncheck them all the favourite items will be removed. For the usabality of app we should not do that :)

**Comment by vyankatesh24** [comment]
According to me on `Done`click changes should be applied and on `Back` click the currently made changes should be discarded Earlier favorite items will be a it is. 

What do you think @Swati4star

**Comment by Swati4star** [comment]
I agree with @vyankatesh24 

Also, add back icon to top left on the action bar (setHomeAsUpEnabled)

**Comment by theabhishekavi** [comment]
@vyankatesh24 @Swati4star made changes as requested by you :)

File: app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java [hunk 2]
```diff
@@ -1,16 +1,122 @@
...
+    }
+
+    @Override
+    public void onBackPressed() {
+        onBackPressedState();
+        super.onBackPressed();
+    }
+
+
+    private void storeInitialState() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java]: addd javadoc style comments to the function

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -93,6 +93,7 @@
     <string name="help">Help</string>
 
     <!-- images to PDF -->
+    <string name="images_to_pdf">Images to PDF</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: add this string for other local files also (in english only)

File: app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java (hunk 2 above)
Comment (theabhishekavi) [reply app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java]: Okay

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (theabhishekavi) [reply app/src/main/res/values/strings.xml]: Done

**Comment by theabhishekavi** [comment]
Done! @vyankatesh24

**Pull request merged**